lib/mercury/fake.rb

```diff
@@ -113,7 +113,7 @@ def roundtrip(msg)
     end
 
     def ensure_queue(source, tag_filter, worker)
-      require_ack = worker != nil
+      require_ack = !worker.nil?
```

Comment:
If you can go based on a truthy value, then you could have just gone with `worker`, however that may mislead your intent for truthyness with something else. 

Another trick could have been `!!worker` if you indeed need a boolean.